app/views/avo/fields/index/_status-field.html.erb

```diff
@@ -0,0 +1,8 @@
+<%= index_field_wrapper field: field do %>
+  <%
+    status = 'success'
+    status = 'fail' if field[:failed_when].include? field[:value].to_sym
+    status = 'load' if field[:loading_when].include? field[:value].to_sym
+  %>
```

Comment:
I think we can leave all of this out and not have to deal with the duplication if we update the `hydrate_field` like so.

```ruby
      def hydrate_field(fields, model, resource, view)
        status = 'success'
        status = 'fail' if @failed_when.include? fields[:value].to_sym
        status = 'load' if @loading_when.include? fields[:value].to_sym

        {
          status: status,
        }
      end
```

Reply:
we don't have the value set there, so that's why we need to compute on the view components

Reply:
we don't? it's in the `fields` (which should be renamed).

Reply:
I've tried to get the value in there as `fields[:value]` and it returned `nil`. Didn't investigate further.

Reply:
This is what `fields` contains:
```ruby
{:id=>:status, :computed=>false, :name=>"Status", :translation_key=>nil, :block=>nil, :component=>"status-field", :required=>false, :readonly=>false, :updatable=>true, :sortable=>false, :nullable=>true, :null_values=>[nil, ""], :computable=>true, :is_array_param=>false, :format_using=>false, :placeholder=>"Status", :help=>nil, :default=>nil, :value=>nil}
```